src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxTree.ParsedSyntaxTree.cs

```diff
@@ -118,6 +118,7 @@ public override CSharpParseOptions Options
                 }
             }
 
+            [Obsolete]
             public override ImmutableDictionary<string, ReportDiagnostic> DiagnosticOptions => _diagnosticOptions;
```

Comment:
We should always return `EmptyDiagnostics` here

Reply:
I can, but this is a pivot point for what level of breaking change we want to take.

At the moment I'd say it's a pretty mild breaking change. For most users who just use editorconfig, there should be no discernable difference. For analyzer authors who actually use the properties provided on the syntax trees, but don't try to construct syntax trees themselves, they will see no options. For users who construct syntax trees manually, and then access the properties on them, they will still see the items they manually added.

We could decide to additionally break the last set of users -- people who construct syntax trees manually, then access the options. I was holding off on that for the first release, basically giving people time to notice the compile-time warnings before changing runtime behavior, but I think the set of people who do this are very small, possibly zero.